assets/src/edit-story/components/panels/stylePreset/colorPresetActions.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+import styled from 'styled-components';
+import { rgba } from 'polished';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app/story';
+import { PatternPropType } from '../../../types';
+import { findMatchingColor } from './utils';
+
+const ActionsWrapper = styled.div`
+  position: absolute;
+  right: 0;
+  left: 0;
+  bottom: 0;
+  text-align: center;
+  border-top: 1px solid rgba(229, 229, 229, 0.2);
+`;
+
+const AddColorPreset = styled.button`
+  background: transparent;
+  border: none;
+  color: ${({ theme }) => rgba(theme.colors.fg.v7, 0.84)};
+  cursor: pointer;
+  padding: 0;
+  line-height: 18px;
+  font-size: 13px;
+`;
+
+function ColorPresetActions({ color }) {
+  const {
+    state: {
+      selectedElements,
+      story: { stylePresets },
+    },
+    actions: { updateStory },
+  } = useStory();
+
+  const { colors, textColors } = stylePresets;
+
+  // @todo This will change with the missing multi-selection handling.
+  const isText =
+    selectedElements.length > 0 &&
+    selectedElements.every(({ type }) => 'text' === type);
+
+  const handleAddColorPreset = useCallback(() => {
+    if (color) {
+      // If match found, don't add.
+      // @todo UX improvement: notify the user/mark the existing color?
```

Comment:
Are there some ordering rules? B/c it might be best to just reorder the matching color to the top and no messaging is really needed then.

Reply:
With the current UX, the new colors would always go as last.

What I meant here with the UX improvement is that if a color is tried to be added which already exists then it would be nice to let the user know somehow why the color wasn't added. Perhaps highlight the existing color somehow among the color presets for example. 

That's mainly a note for me for when continuing the work in a follow-up PR, will check with Sam today.

Reply:
Yep. In this case, absolutely correct - it'd be good to highlight this somehow. Or we could even always highlight the entered color: whether it's a new one or existing. Let's file an issue on Sam for this.